### ISSUE TITLE
Error parsing the result of /device_updates/<uuid>/queued

### ISSUE THREAD

**Issue by cajun-rat** [advancedtelematic/sota_client_cpp#69]
The data::UpdateRequest::fromJson function assumes that the 'status' enum is an integer, but the actual output of the SOTA Server is an enum containing text strings like "Pending".

Here is a failing test case, derived from an actual interaction with the SOTA Server:

```
BOOST_AUTO_TEST_CASE(UpdatesReceived_parse) {
  std::string json(
      "[ "
      "{\n"
      "   \"createdAt\" : \"2017-01-05T08:43:40.685Z\",\n"
      "		\"installPos\" : 0,\n"
      "		\"packageId\" : \n"
      "		{\n"
      "			\"name\" : \"treehub-ota-raspberrypi3\",\n"
      "			\"version\" : "
      "\"15f13e2582f29d2218f88adada52ff043d6e9596b7cea288321ed91e275a1768\"\n"
      "		},\n"
      "		\"requestId\" : \"06d64e46-cb25-4d76-b62e-4341b6944d07\",\n"
      "		\"status\" : \"Pending\",\n"
      "		\"updatedAt\" : \"2017-01-05T08:43:40.685Z\"\n"
      "	}\n"
      "]");

  data::UpdateRequest request(data::UpdateRequest::fromJson(json));
}
```
The value of 'status' can be either "Pending", "InFlight", "Canceled", "Failed" or "Finished".  The server won't send all of these. I will check with @jerrytrieu which are possible.

The server-side definitions of this are here:

https://github.com/advancedtelematic/rvi_sota_server/blob/master/core/src/main/scala/org/genivi/sota/core/data/client/PendingUpdateRequest.scala

https://github.com/advancedtelematic/rvi_sota_server/blob/master/core/src/main/scala/org/genivi/sota/core/data/UpdateRequest.scala#L66

**Comment by taheris** [comment]
The rust definition is:

```
pub enum UpdateRequestStatus {
    Pending,
    InFlight,
    Canceled,
    Failed,
    Finished
}
```

There are decodable/encodable traits that will parse an enum value to and from a string. Internally the `UpdateRequestStatus` type may be represented as an integer, but this is an implementation detail and the true representation (e.g. for any external interaction) is a string.

**Comment by cajun-rat** [comment]
Fixed in #72
